types/option.go

```diff
@@ -185,6 +185,11 @@ type GetObjectOption struct {
 	Range string `url:"-" header:"Range,omitempty"` // support for downloading partial data
 }
 
+type GetChallengeInfoOptions struct {
```

Comment:
Could you add some comments to explain the priority of 
- Routing by `Endpoint` 
- Routing by  `SPAddress` 
- Auto-Routing? 


E.g. : `Endpoint` > `SPAddress` > Auto-Routing

Reply:
add an comment on the interface defination